src/topic_monitor.cpp

```diff
@@ -37,12 +37,14 @@ wcstring generation_list_t::describe() const {
     return result;
 }
 
-binary_semaphore_t::binary_semaphore_t() : sem_ok_(false) {
+#ifdef __linux__
+binary_semaphore_t::binary_semaphore_t() {
     // sem_init always fails with ENOSYS on Mac and has an annoying deprecation warning.
     // On BSD sem_init uses a file descriptor under the hood which doesn't get CLOEXEC (see #7304).
     // So use fast semaphores on Linux only.
-#ifdef __linux__
     sem_ok_ = (0 == sem_init(&sem_, 0, 0));
+#else
+binary_semaphore_t::binary_semaphore_t() : sem_ok_(false) {
```

Comment:
Was this automated? I find the original much easier to follow: here's the default value, here's the override for this case. I get that it's a redundant assignment, but realistically how much does that cost?

An alternative I would also be fine with:

```cpp
binary_semaphore_t::binary_semaphore_t() {
    // comment here
#ifdef __linux__
    sem_ok_ = (0 == sem_init(&sem_, 0, 0));
#else 
    sem_ok = false;
#endif 
}
```

Repeating the declaration twice seems like it's asking for trouble, e.g. forgetting to make changes to one of the branches when you change the other.

Reply:
This was my solution to avoiding the warning.